### DISCUSSION TITLE
Can I set K to prevent memory overflow

### DISCUSSION THREAD

**Discussion by xug15** [ablab/spades#1121] [Q&A]
### Is your feature request related to a problem? Please describe. For generic questions use Q&A section in the Discussions forum above.

That's great tools for us, I am associate professor from CAU.
When I use spades to assembly metagenome sample from maize, the always :Error in malloc(): out of memory. Requested: 94287658224, active: 32942063616
I notice the K21 and K22 both were finished, but the K 55 always out of memory.
So can I just use K33 to assembly metagenome? I think that one sample have 4 replicate illumina fastq.gz. One is out of memory, how can I use 4 replicate to use all data in one sample?

**Comment by asl** [comment]
Hello

Your very first step is to use newer version of SPAdes (3.15.5), as you're using something ancient.

Usage of K33 will not help as the last stage is always the main regardless of k-mer length. K33 being last will use more RAM due to more tangled and complex assembly graph.
